spec/syntax.md

```diff
@@ -691,6 +697,39 @@ _Markup_ _placeholders_ can appear in any order without making the _message_ inv
 However, specifications or implementations defining _markup_ might impose requirements
 on the pairing, ordering, or contents of _markup_ during _formatting_.
 
+## Attributes
+
+An **_<dfn>attribute</dfn>_** is a key with an optional value that
+is attached to the _expression_ or _markup_ as a whole.
```

Comment:
This is what we intend to use it for, but I would probably not say it like this. Note that `key` should be `identifier` and that "the expression" is slightly quirky. Perhaps:

```suggestion
An **_<dfn>attribute</dfn>_** is an identifier with an optional value
that appears in an _expression_ or in _markup_.
```